src/layout/constraint.rs

```diff
@@ -2,97 +2,204 @@ use std::fmt::{self, Display};
 
 use itertools::Itertools;
 
-/// A constraint that can be applied to a layout
+/// A constraint that defines the size of a layout element.
 ///
-/// Constraints are used to define the size of a layout. They can be used to define a fixed size, a
-/// percentage of the available space, a ratio of the available space, or a minimum or maximum size.
+/// Constraints can be used to specify a fixed size, a percentage of the available space, a ratio of
+/// the available space, a minimum or maximum size or a fill for a layout element.
 ///
 /// Relative constraints (percentage, ratio) are calculated relative to the entire space being
-/// split, not the space available after applying the more fixed constraints (min, max, length).
+/// divided, rather than the space available after applying more fixed constraints (min, max,
+/// length).
 ///
 /// # Examples
 ///
-/// `Constraint` has some helper methods to create lists of constraints from anything that can be
-/// converted into an iterator of u16s ((u16, u16) for ratios).
+/// `Constraint` provides helper methods to create lists of constraints from various input formats.
 ///
 /// ```rust
 /// # use ratatui::prelude::*;
-/// // a fixed layout
+/// // Create a fixed layout with specified sizes for each element
 /// let constraints = Constraint::from_lengths([10, 20, 10]);
 ///
-/// // a centered layout
+/// // Create a centered layout using ratio or percentage constraints
 /// let constraints = Constraint::from_ratios([(1, 4), (1, 2), (1, 4)]);
 /// let constraints = Constraint::from_percentages([25, 50, 25]);
 ///
-/// // a centered layout with a minimum size
+/// // Create a centered layout with a minimum size constraint for specific elements
 /// let constraints = Constraint::from_mins([0, 100, 0]);
 ///
-/// // a sidebar layout specifying maximum sizes of the columns
+/// // Create a sidebar layout specifying maximum sizes for the columns
 /// let constraints = Constraint::from_maxes([30, 170]);
 /// ```
 #[derive(Debug, Clone, Copy, Eq, PartialEq, Hash)]
 pub enum Constraint {
-    /// Apply a percentage to a given amount
+    /// Applies a percentage of the available space to the element
     ///
-    /// Converts the given percentage to a f32, and then converts it back, trimming off the decimal
-    /// point (effectively rounding down)
+    /// Converts the given percentage to a floating-point value, then back to an integer, trimming
+    /// off the decimal point (effectively rounding down).
```

Comment:
I think this description might be wrong. It applied when we used the `apply` method (which did truncate the area), but that changed when we moved the conversion to an area inside the split method, so splitting 3 into 50/50 works fine - it rounds up for the first element and down for the second.

```rust
Constraint::Percentage(p) => {
    let percent = f64::from(p) / 100.00;
    solver.add_constraint(element.size() | EQ(STRONG) | (area_size * percent))?;
}
```